apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/ExecuteHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.V1.Logging;
+
+namespace Google.Cloud.Spanner
+{
+    internal static class ExecuteHelper
+    {
+        internal static void WithErrorTranslationAndProfiling(Action t, string name)
+        {
+            try
+            {
+                Stopwatch sw = null;
+                if (Logger.LogPerformanceTraces)
+                    sw = Stopwatch.StartNew();
+                t();
+                if (sw != null)
+                    Logger.LogPerformanceCounterFn($"{name}.Duration", x => sw.ElapsedMilliseconds);
+            }
+            catch (Exception e)
+            {
+                SpannerException translatedException;
+                if (SpannerException.TryTranslateRpcException(e, out translatedException))
+                    throw translatedException;
+                throw;
+            }
+        }
+
+        internal static async Task WithErrorTranslationAndProfiling(Func<Task> t, string name)
```

Comment:
Rewrite this to just call the other overload (using a `Task<int>` or similar)? May not be worth it.